wire-ios-data-model/Source/Model/Message/ZMAssetClientMessage+Ephemeral.swift

```diff
@@ -65,16 +65,18 @@ extension ZMAssetClientMessage {
     }
 
     @discardableResult @objc public override func startDestructionIfNeeded() -> Bool {
-
-        if imageMessageData != nil && !hasDownloadedFile {
-            return false
-        } else if fileMessageData != nil
+        let imageNotDownloaded = imageMessageData != nil && !hasDownloadedFile
+        let fileNotUploadedNorUploaded = fileMessageData != nil
             && underlyingMessage?.assetData?.hasUploaded == false
-            && underlyingMessage?.assetData?.hasNotUploaded == false {
+            && underlyingMessage?.assetData?.hasNotUploaded == false
+        let otherSenderCheck = sender == nil || (sender?.isSelfUser == false && managedObjectContext?.zm_isUserInterfaceContext == true)
```

Comment:
while we're at it, can we improve the name of this one?

Reply:
Changed :)